src/transformers/modeling_utils.py

```
@@ -3972,6 +3977,7 @@ def from_pretrained(
         generation_config = kwargs.pop("generation_config", None)
         gguf_file = kwargs.pop("gguf_file", None)
         tp_plan = kwargs.pop("tp_plan", None)
+        tp_size = kwargs.pop("tp_size", None)
```

let's raise an error if `tp_size` was set but not `tp_plan`